src/game.cpp

```diff
@@ -8717,19 +8742,20 @@ void game::butcher()
             kmenu.addentry( MULTIBUTCHER, true, 'b', _( "Butcher everything" ) );
         }
         if( disassembles.size() > 1 ) {
-            int time_to_disassemble = 0;
-            int time_to_disassemble_all = 0;
+            int time_to_disassemble_once = 0;
+            int time_to_disassemble_recursive = 0;
             for( const auto &stack : disassembly_stacks ) {
                 const int time = recipe_dictionary::get_uncraft( stack.first->typeId() ).time_to_craft_moves(
                                      get_player_character() );
-                time_to_disassemble += time;
-                time_to_disassemble_all += time * stack.second;
+                time_to_disassemble_once += time * stack.second;
+                time_to_disassemble_recursive += get_recursive_disassemble_moves( get_player_character(),
+                                                 stack.first->typeId(), stack.second );
```

Comment:
Using item::typeId here will always calculate the time for the default recipe, not for the actual components the item might be made of. So for example a light frame made from short vines will calculate as if it was made from short ropes instead. Vines are not disassemblable, while short ropes disassemble down to threads. Things with a lot of disassembly depth also make this really slow, because every single resulting item will be calculated (1 light frame -> 3 short ropes -> 18 long strings -> 104 short strings).

I'd suggest moving the method to item and then calling it directly on the item to get the duration for the actual components. I'm not qualified to make suggestions to improve performance, though.

Reply:
Thank you for your feedback, I moved the method to item and modified the signature. It should now take the actual components of the item into account, I added a test case for it.
Unfortunately, it seems there's a bug of some sort. The last time I tried, it displayed ``21 hours and 53 minutes`` instead of ``21 hours and 50 minutes`` when disassembling the light wooden frame made of short rope. I'm still tracking the issue.